components/odh-notebook-controller-webhook/main.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"flag"
+	"time"
+
+	"github.com/opendatahub-io/kubeflow/components/odh-notebook-controller/controllers"
+	"go.uber.org/zap/zapcore"
+	k8sclient "k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlruntime "sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	webhook "sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+var (
+	webhookLog = ctrl.Log.WithName("webhook")
+)
+
+func main() {
+
+	var metricsAddr string
+	var webhookPort int
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080",
+		"The address the metric endpoint binds to.")
+	flag.IntVar(&webhookPort, "webhook-port", 8443,
+		"Port that the webhook server serves at.")
+
+	opts := zap.Options{
+		Development: true,
+		TimeEncoder: zapcore.TimeEncoderOfLayout(time.RFC3339),
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	// Setup logger
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	var oauthProxyImage string
+	flag.StringVar(&oauthProxyImage, "oauth-proxy-image", controllers.OAuthProxyImage,
+		"Image of the OAuth proxy sidecar container.")
+
+	config, err := ctrlruntime.GetConfig()
+	if err != nil {
+		webhookLog.Error(err, "Error creating the config object")
+	}
+
+	kc, err := k8sclient.NewForConfig(config)
+	if err != nil {
+		webhookLog.Error(err, "Failed to initialize Kubernetes client")
+	}
+
+	// Setup notebook mutating webhook
+	hookServer := webhook.Server{
+		Host: metricsAddr,
+		Port: webhookPort,
```

Comment:
This needs to be updated to include webhook cert directory.
- https://github.com/opendatahub-io/kubeflow/blob/master/components/odh-notebook-controller/Makefile#L14
- Updates in manifests - https://github.com/opendatahub-io/kubeflow/tree/master/components/odh-notebook-controller/config/webhook